python-avd/pyavd/_eos_designs/shared_utils/filtered_tenants.py

```diff
@@ -182,16 +182,61 @@ def is_forced_vrf(self: SharedUtils, vrf: dict) -> bool:
 
         return vrf["name"] in (self.get_switch_fact("uplink_switch_vrfs", required=False) or [])
 
-    def filtered_vrfs(self: SharedUtils, tenant: dict) -> list[dict]:
+    @cached_property
+    def local_vrfs(self: SharedUtils) -> list:
         """
-        Return sorted and filtered vrf list from given tenant.
+        Return the list of vrfs locally defined on this switch.
 
-        Filtering based on svi tags, l3interfaces, loopbacks or self.is_forced_vrf() check.
-        Keys of VRF data model will be converted to lists.
+        Ex. ["default", "prod"]
         """
-        filtered_vrfs = []
+        if not self.network_services_l3:
+            return []
+
+        vrfs = {vrf["name"] for vrf in self._build_vrfs[0]}
+        return natural_sort(vrfs)
+
```

Comment:
We could make this dicts keyed by tenant to make it easier to parse later and to work around the issue you pointed out.

Reply:
done

Reply:
though not sure how to walk around the issue I mentionned